### PR TITLE
Make shell script safer

### PR DESCRIPTION
**Pull request by szepeviktor**
and more readable

Please consider running shellcheck in CI

### CODE REVIEW

**Comment by bartlomieju** [comment]
CC @MarkTiedemann please review

**Comment by MarkTiedemann** [comment]
I initially added `shellcheck` to the CI with https://github.com/denoland/deno_install/pull/23. I think it got removed when migrating to Github Actions.

In any case, I already have a PR open for re-adding it, see: https://github.com/denoland/deno_install/pull/121

I think this MR can be closed as a duplicate. @szepeviktor, please feel free to suggest changes to #121.

**Comment by szepeviktor** [comment]
@MarkTiedemann :) This PR would be my suggestion :)

File: install.sh [hunk 1]
```diff
@@ -4,23 +4,25 @@
 
 set -e
 
-case $(uname -s) in
+case "$(uname -s)" in
```
Comment (MarkTiedemann) [inline install.sh:7]: This change does not prevent word splitting. Please revert.

File: install.sh [hunk 2]
```diff
@@ -4,23 +4,25 @@
...
 
-if [ $(uname -m) != "x86_64" ]; then
+if [ "$(uname -m)" != "x86_64" ]; then
 	echo "Unsupported architecture $(uname -m). Only x64 binaries are available."
 	exit
 fi
 
-if [ $# -eq 0 ]; then
-	deno_asset_path=$(
+if [ "$#" -eq 0 ]; then
```
Comment (MarkTiedemann) [inline install.sh:17]: This change doesn't prevent word splitting either. `$#` is guaranteed to be a positive natural number.

Please revert.

File: install.sh (hunk 1 above)
Comment (szepeviktor) [reply install.sh]: Done.

File: install.sh [hunk 3]
```diff
@@ -4,23 +4,25 @@
...
+	deno_asset_path="$(
 		command curl -sSf https://github.com/denoland/deno/releases |
 			command grep -o "/denoland/deno/releases/download/.*/deno-${target}\\.zip" |
 			command head -n 1
-	)
-	if [ ! "$deno_asset_path" ]; then exit 1; fi
+	)"
+	if [ ! "$deno_asset_path" ]; then
+		exit 1
+	fi
```
Comment (MarkTiedemann) [inline install.sh:25]: Formatting, please revert.

There's another PR which adresses error messages: https://github.com/denoland/deno_install/pull/122

If you want to improve error messages, feel free to add to that PR instead. :)

File: install.sh (hunk 2 above)
Comment (szepeviktor) [reply install.sh:17]: I do not pick and choose. My strategy is to quote every variable.

File: install.sh [hunk 4]
```diff
@@ -4,23 +4,25 @@
...
-if [ $(uname -m) != "x86_64" ]; then
+if [ "$(uname -m)" != "x86_64" ]; then
 	echo "Unsupported architecture $(uname -m). Only x64 binaries are available."
 	exit
 fi
 
-if [ $# -eq 0 ]; then
-	deno_asset_path=$(
+if [ "$#" -eq 0 ]; then
+	deno_asset_path="$(
```
Comment (MarkTiedemann) [inline install.sh:18]: Assigning `deno_asset_path` does not need to be quoted either. Please revert.

File: install.sh [hunk 5]
```diff
@@ -4,23 +4,25 @@
...
 set -e
 
-case $(uname -s) in
+case "$(uname -s)" in
 Darwin) target="x86_64-apple-darwin" ;;
 *) target="x86_64-unknown-linux-gnu" ;;
 esac
 
-if [ $(uname -m) != "x86_64" ]; then
+if [ "$(uname -m)" != "x86_64" ]; then
```
Comment (MarkTiedemann) [inline install.sh:12]: This change is fine. :) It is already adressed by https://github.com/denoland/deno_install/pull/121, however.

File: install.sh (hunk 3 above)
Comment (szepeviktor) [reply install.sh:25]: This revert would make this PR an empty string :)

**Comment by MarkTiedemann** [comment]
@szepeviktor Sorry for the rather harsh comments.

Again, feel free to add to https://github.com/denoland/deno_install/pull/121.

There's also a few open issues. If you need help with any of them, please let me know. I'd be happy to support if necessary.

Hope you have a good day. :)

**Pull request closed without merging**